tensorflow_probability/python/experimental/mcmc/covariance_reducer.py

```diff
@@ -34,6 +37,10 @@
 ]
 
 
+CovarianceReducerState = collections.namedtuple(
+    'CovarianceReducerState', 'init_structure, cov_state')
```

Comment:
It's not a huge deal, but could you extract `init_structure` from `cov_state`, somehow? It's a little bit wasteful to carry `init_structure` which isn't changing between steps.

Reply:
I tried playing around with it for the good part of the morning, but extracting from `cov_state` runs into the issue of some arbitrarily nested structure. In say the case of a tuple of dictionaries of reducer states, without keeping some reference to the structure of that initial chain state, we then don't know exactly how to extract a Tensor that would help. I'm sure there's a workaround, but that would probably lead to much more complex code.

Reply:
Would it not just be `map_structure(lambda cs: cs.mean, cov_state)`? I'm confused when a case would happen where the structure of `cov_state` wouldn't match `init_structure`.

Reply:
Wouldn't that actually go as deep as possible until it hits `cov_state`'s Tensors? It would be calling something like Tensor.mean (which doesn't work). We still need some `map_structure_up_to` to specify the shallow tree. However, in `finalize` we can't do that because all we have is the final reducer state.

Reply:
Ah yes, of course! You're right. I guess let's keep the `init_structure` for now, and we can think of ways of getting rid of it later.